core/src/main/scala/org/scalatra/servlet/RichServletContext.scala

```diff
@@ -222,5 +222,17 @@ case class RichServletContext(sc: ServletContext) extends AttributesMap {
 
   def contextPath: String = sc.getContextPath
 
+  val generatedUrlsKey = "org.scalatra.scalate.generatedUrls"
+
+  def addGeneratedUrls(servletClassName: String, routes: Map[String, Route]) = {
+    sc.setAttribute(generatedUrlsKey, generatedUrls + (servletClassName -> routes))
+  }
+
+  def generatedUrls = {
+    Option(sc.getAttribute(generatedUrlsKey))
+      .getOrElse(Map.empty[String, Map[String, Route]])
+      .asInstanceOf[Map[String, Map[String, Route]]]
+  }
```

Comment:
I think that these code should be in `ScalateUrlGeneratorSupport` side because `RichServletContext` shouldn't know something about `ScalateUrlGeneratorSupport`.

Reply:
Good point!